did-tezos/src/lib.rs

```diff
@@ -130,16 +201,162 @@ impl DIDMethod for DIDTz {
+            assertion_method: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            verification_method: Some(vec![VerificationMethod::Map(VerificationMethodMap {
+                id: String::from(vm_didurl.clone()),
+                type_: proof_type.to_string(),
+                controller: did.to_string(),
+                blockchain_account_id: Some(format!("{}@tezos:{}", address.to_string(), network)),
+                ..Default::default()
+            })]),
+            ..Default::default()
+        }
+    }
+
+    async fn tier2_resolution(did: &str, address: &str, network: &str) -> Result<Service> {
+        let did_manager = explorer::retrieve_did_manager(address, network).await?;
+        Ok(explorer::execute_service_view(did, &did_manager, network).await?)
+    }
+
+    async fn tier3_updates(
+        &self,
+        prefix: &str,
+        doc: &mut Document,
+        updates: Updates,
+        public_key: Option<String>,
+    ) -> Result<()> {
+        match updates {
+            Updates::SignedIetfJsonPatch(patches) => {
+                for jws in patches {
+                    let mut doc_json = serde_json::to_value(&mut *doc)?;
+                    let curve = prefix_to_curve_type(prefix)
+                        .ok_or(anyhow!("Unsupported curve."))?
+                        .0
+                        .to_string();
+                    let jwk = match prefix {
+                        "tz1" => {
+                            let pk = match public_key {
+                                Some(ref p) => {
+                                    PublicKey::from_base58check(&p.clone()).unwrap().as_ref()[..]
+                                        .to_vec()
+                                }
+                                None => return Err(anyhow!("Need public key for signed patches")),
+                            };
+                            JWK {
+                                params: Params::OKP(OctetParams {
+                                    curve,
+                                    public_key: Base64urlUInt(pk),
+                                    private_key: None,
+                                }),
+                                public_key_use: None,
+                                key_operations: None,
+                                algorithm: None,
+                                key_id: None,
+                                x509_url: None,
+                                x509_thumbprint_sha1: None,
+                                x509_certificate_chain: None,
+                                x509_thumbprint_sha256: None,
+                            }
+                        }
+                        "tz2" => {
+                            // TODO use tezedge_client when it handles tz2
+                            let pk = match public_key {
+                                Some(ref p) => p.from_base58check().unwrap()[4..].to_vec(),
```

Comment:
Could panic

Reply:
My bad, copy pasted from tests :disappointed: Resolved in 6eb9996